### PR TITLE
Split test to help work around travis timeout. They are super fast here.

### PR DESCRIPTION
**Pull request by nouiz**
fix https://github.com/Theano/Theano/issues/4610

### CODE REVIEW

**Comment by abergeron** [comment]
Ok when travis passes.

**Comment by nouiz** [comment]
I added other fix in this PR to help travis. I canceled the old build.

File: theano/gof/opt.py [hunk 1]
```diff
@@ -315,17 +315,17 @@ def print_profile(stream, prof, level=0):
...
+        for (opt, nb_n) in zip(opts, nb_nodes):
             if hasattr(opt, "__name__"):
                 name = opt.__name__
             else:
                 name = opt.name
             idx = opts.index(opt)
             ll.append((name, opt.__class__.__name__,
-                       idx))
-        lll = sorted(zip(prof, ll, nb_nodes), key=lambda a: a[0])
+                       idx) + nb_n)
```
Comment (abergeron) [inline theano/gof/opt.py]: Is nb_n a tuple here?  Otherwise this might not work.

**Comment by abergeron** [comment]
Ok modulo my comment.

File: theano/gof/opt.py (hunk 1 above)
Comment (nouiz) [inline theano/gof/opt.py]: yes, it is a tuple.

**Comment by nouiz** [comment]
travis failed due to timeout in FAST_COMPILE in the tests that I split here. I don't know what is going on, locally, it take just a few seconds... I'll merge as the master will be better with this PR merged.

**Pull request merged**